### PR TITLE
Gjoranv/split parent

### PR DESCRIPTION
**Pull request by gjoranv**
This introduces a new parent above com.yahoo.vespa:parent, namely `container-dependency-versions`. The dependencyManagement in this pom has the exact same deps as those that are whitelisted in container-dependencies-enforcer, and constitutes the set of provided 3rd party deps from the container.

`container-dependency-versions` is planned to be the new parent of the tenant:base pom.

In addition, I have split the vespa:parent pom from the aggregator root pom. This is the recommended pom structure for maven, and was necessary for the old parent to inherit container-dependency-versions.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -44,8 +44,10 @@ $top/dist/getversion.pl -M $top > $top/dist/vtag.map
 # The 'default' mode also builds some modules needed by C++ code.
 # The 'full' mode also builds modules needed by C++ tests.
 
-# must install parent pom first:
+# must install parent poms first:
 echo "Downloading all dependencies. This may take a few minutes with an empty Maven cache."
+cd container-dependency-versions && mvn_install
+cd ../parent && mvn_install && cd ..
```
Comment (aressem) [inline bootstrap.sh]: I would recommend expressing the two above lines like this:
```
(
  cd container-dependency-versions 
  mvn_install
)
(
  cd parent
  mvn_install
)
```
The code inside the parenthesis is executed in its own context and whatever you do inside wont affect the working directory of the rest of the script.

**Review by aressem — CHANGES_REQUESTED** [review]
See comment about changing directories.

**Comment by bjorncs** [comment]
Note: Travis-CI build failed after last commit.

**Review by bjorncs — APPROVED** [review]

**Review by aressem — APPROVED** [review]

**Comment by gjoranv** [comment]
The failure on travis-ci was due to what I assume is an unstable test in clustercontroller-core..

**Comment by arnej27959** [comment]
you'll need to account for this in the vespa-yahoo repo to ensure factory builds work; the new poms need to be "mvn install"ed and packaged into yinst packages from somewhere.

**Pull request merged**